atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/expectations/ExpectationsAwareTransaction.java

```diff
@@ -16,16 +16,19 @@
 
 package com.palantir.atlasdb.transaction.api.expectations;
 
-import com.google.common.collect.ImmutableMap;
-import com.palantir.atlasdb.keyvalue.api.TableReference;
-import org.immutables.value.Value;
+import com.palantir.atlasdb.transaction.api.Transaction;
 
-/**
- * Tracked data for an individual transaction.
- */
-@Value.Immutable
-public interface ExpectationsStatistics {
-    long transactionAgeMillis();
+public interface ExpectationsAwareTransaction extends Transaction {
+    long getAgeMillis();
+
+    /**
+     * Returns a point-in-time snapshot of transaction read information.
+     */
+    TransactionReadInfo getReadInfo();
 
-    ImmutableMap<TableReference, TransactionReadInfo> readInfoByTable();
+    /**
+     * Update TEX data collection metrics for post-mortem transactions.
+     * Invoke only after the transaction committed or aborted.
```

Comment:
What happens if they invoke before? Here be dragons?

Reply:
No guarantees, we will just get marginally worse metris (depending how the calling incorrectness is). Also, adding in the javadoc that this should only be called exactly once